src/packagedcode/windows_helpers.py

```diff
@@ -0,0 +1,311 @@
+        registry_path=registry_path
+    )
+    if not name_key_entry:
+        return
+
+    # Collect Package information and create Package if we have a valid Package name
+    for entry in registry_hive.recurse_subkeys(name_key_entry):
+        package_info = {}
+        for entry_value in entry.values:
+            if entry_value.name == 'DisplayName':
+                package_info['DisplayName'] = entry_value.value
+            if entry_value.name == 'DisplayVersion':
+                package_info['DisplayVersion'] = entry_value.value
+            if entry_value.name == 'InstallLocation':
+                package_info['InstallLocation'] = entry_value.value
+            if entry_value.name == 'Publisher':
+                package_info['Publisher'] = entry_value.value
+            if entry_value.name == 'URLInfoAbout':
+                package_info['URLInfoAbout'] = entry_value.value
+
+        name = package_info.get('DisplayName')
+        if not name:
+            continue
+
+        version = package_info.get('DisplayVersion')
+        install_location = package_info.get('InstallLocation')
+        publisher = package_info.get('Publisher')
+
+        parties = []
+        if publisher:
+            parties.append(
+                models.Party(
+                    type=None,
+                    role='publisher',
+                    name=publisher
+                )
+            )
+
+        extra_data = {}
+        if install_location:
+            extra_data['install_location'] = install_location
+
+        yield InstalledWindowsProgram(
+            name=name,
+            version=version,
+            parties=parties,
+            extra_data=extra_data
+        )
+
+
+def reg_parse(location):
+    for installed_program in report_installed_programs(location):
+        yield installed_program
+    for installed_dotnet in report_installed_dotnet_versions(location):
+        yield installed_dotnet
+
+
+def get_installed_programs(root_dir):
+    """
+    Yield InstalledWindowsProgram objects for every detected installed program
+    from Windows registry files in known locations
+    """
+
+    # These paths are relative to a Windows docker image layer root directory
+    software_delta_loc = os.path.join(root_dir, 'Hives', 'Software_Delta')
+    SOFTWARE_reg_file_loc = os.path.join(root_dir, 'Files/Windows/System32/config/SOFTWARE')
+    software_registry_locations = [
+        software_delta_loc,
+        SOFTWARE_reg_file_loc
+    ]
+    for software_registry_loc in software_registry_locations:
+        if not os.path.exists(software_registry_loc):
+            continue
+        for installed_program in reg_parse(software_registry_loc):
+            installed_program.populate_installed_files(root_dir)
+            yield installed_program
+
+
+@attr.s()
+class InstalledWindowsProgram(models.Package):
+    filetypes = ('SOFTWARE',)
+    mimetypes = ('application/octet-stream',)
+    default_type = 'windows-program'
```

Comment:
What about just "windows" ? We need to find a good set of types here and for msi and the related win_pe and possibly MS Cabinets.

Reply:
I left this as `windows-program`